src/metabase/search.clj

```diff
@@ -35,6 +35,25 @@
                     search.impl/in-place))
     :in-place search.impl/in-place))
 
+(defn supports-index?
+  "Does this instance support a search index, e.g. has the right kind of AppDb"
+  []
+  (is-postgres?))
```

Comment:
:joy:

Reply:
All these checks are a bit sad, thinking it might be time to just add a multimethod already.

Reply:
should the check for the env flag be done here, too? It'd be nice to keep that a module gate thing and not leak it out.

Reply:
@qnkhuat That's a reasonable question. I prefer to check that sort of thing as close to the periphery of the system as possible, to save stubbing in tests and REPL usage, but it does increase the likelihood of forgetting the check somewhere.

I'm incline to leave as is for now, as reworking with multi methods looks imminent.